src/Appwrite/Auth/OAuth2/Dailymotion.php

```diff
@@ -0,0 +1,223 @@
+                'scope' => \implode(' ', $this->getScopes())
+            ]);
+
+        return $url;
+    }
+
+    /**
+     * @param string $code
+     *
+     * @return array
+     */
+    protected function getTokens(string $code): array
+    {
+        if (empty($this->tokens)) {
+            $this->tokens = \json_decode($this->request(
+                'POST',
+                $this->endpoint . '/oauth/token',
+                ["Content-Type: application/x-www-form-urlencoded"],
+                \http_build_query([
+                    'grant_type' => 'authorization_code',
+                    "client_id" => $this->appID,
+                    "client_secret" => $this->appSecret,
+                    "redirect_uri" => $this->callback,
+                    'code' => $code,
+                    'scope' => \implode(' ', $this->getScopes()),
+                ])
+            ), true);
+        }
+        return $this->tokens;
+    }
+
+
+    /**
+     * @param string $refreshToken
+     *
+     * @return array
+     */
+    public function refreshTokens(string $refreshToken): array
+    {
+
+        $this->tokens = \json_decode($this->request(
+            'POST',
+            $this->endpoint . '/oauth/token',
+            ['Content-Type: application/x-www-form-urlencoded'],
+            \http_build_query([
+                'grant_type' => 'refresh_token',
+                'refresh_token' => $refreshToken,
+                'client_id' => $this->appID,
+                'client_secret' => $this->appSecret,
+            ])
+        ), true);
+
+        if (empty($this->tokens['refresh_token'])) {
+            $this->tokens['refresh_token'] = $refreshToken;
+        }
+
+
+        return $this->tokens;
+    }
+
+    /**
+     * @param string $accessToken
+     *
+     * @return string
+     */
+    public function getUserID(string $accessToken): string
+    {
+        $user = $this->getUser($accessToken);
+
+        $userId = $user['id'] ?? '';
+
+        return $userId;
+    }
+
+    /**
+     * @param string $accessToken
+     *
+     * @return string
+     */
+    public function getUserEmail(string $accessToken): string
+    {
+        $user = $this->getUser($accessToken);
+
+
+        $userEmail = $user['email'] ?? '';
+
+        return $userEmail;
+    }
+
+    /**
+     * Check if the OAuth email is verified
+     *
+     * @link https://developers.dailymotion.com/api/#user-fields
+     *
+     * @param string $accessToken
+     *
+     * @return bool
+     */
+    public function isEmailVerified(string $accessToken): bool
+    {
+        $user = $this->getUser($accessToken);
+
+        if ($user['verified'] ?? false) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $accessToken
+     *
+     * @return string
+     */
+    public function getUserName(string $accessToken): string
+    {
+        $user = $this->getUser($accessToken);
+
+
+        $username = $user['username'] ?? '';
```

Comment:
Let's use first name and last name here.

Reply:
you can also pass `fullname` in the fields . I am trying to avoid `first_name` here cause it can be null sometimes as in case of mine.
```json
{  "first_name": null,  
   "last_name": "2002bishwajeet",  
   "fullname": "2002bishwajeet",  
   "verified": false,  
   "email": "bishwajeet.techmaster@gmail.com",  
   "id": "x2nz1j4"
}
So I replace this with `fullname` .
```

Reply:
Yes, please replace with `fullname`